Add Entry General: Vim (#593)
Codecademy/docs: content/general/concepts/vim/vim.md

---
Title: 'Vim'
Description: 'Vim is a free, open source, and customizable text editor for Linux.'
Subjects:
  - 'Bash/Shell'
  - 'Developer Tools'
Tags:
  - 'Vim'
  - 'Linux'
  - 'Developer Tools'
CatalogContent:

>   - 'paths/computer-science'

Reviewer: ```suggestion
  - 'learn-the-command-line'
  - 'paths/computer-science'
```

The `CatalogContent` should always include two links:

 - One free course slug.
 - One paid `paths/` slug.

---

[Vim](https://www.vim.org/) is a free, open source, and customizable text editor that can be run on Windows, Mac, Unix, Linux, or Android. It can be run directly from the command line interface or from it's GUI interface. It is based on 'vi', another text editor that was created for the unix operating system. Many plugins are available to add to the many built in features. Developers will find it useful for working on source code or scripting files. Vim is also very convenient for editing configuration files when working from the terminal.

## Features

- Text completion
- File comparison and merging
- Search for regular expressions
- Can perform complicated tasks using only commands
- Open multiple tabs
- Create macros
- Save sessions

## Launching Vim

> 

Reviewer: ```suggestion

Vim can be run directly from the command line interface or from its GUI.

```

> from its GUI interface

Could we include a small screenshot of this, as well?

```bash
vim filename.txt
```

> 

Reviewer: Maybe a short sentence or two below this snippet explaining what the command is doing? Is it opening a new Vim window with the contents of `filename.txt`?

## Vim Screenshot

> ![Vim screenshot](/media/vim.png)

Reviewer: ```suggestion
![Vim screenshot](https://raw.githubusercontent.com/Codecademy/docs/main/media/vim.png)
```

Reviewer: Also could we switch this image with what the screen would look like after running just `vim`?

<img width="682" alt="Screen Shot 2022-03-17 at 9 54 27 AM" src="https://user-images.githubusercontent.com/23144845/158824185-86d1cb3d-2e0c-4f05-aa89-90d6051149e3.png">

---
Title: 'Vim'

> Description: 'Vim is a free, open source, and customizable text editor for Linux.'

Reviewer: ```suggestion
Description: 'Vim is a free, open source text editor that is customizable text and available on many operating systems.'
```

Tags:
  - 'Vim'
  - 'Linux'
  - 'Developer Tools'
CatalogContent:
  - 'learn-the-command-line'
  - 'paths/computer-science'
---

[Vim](https://www.vim.org/) is a text editor that is free, [open-source](https://www.codecademy.com/resources/docs/general/open-source), and based on the `vi` editor that was originally created for the Unix operating system. However, it can be run on other systems such as Windows, Mac, Linux, or Android.

Many plugins are available to add to the many features built into Vim. Developers will find it useful for working on source code or scripting files. Vim is ideal for editing configuration files when working from the [terminal](https://www.codecademy.com/resources/docs/general/terminal).

## Features

This section describes some of the important features of the Vim editor.

- It has text completion for faster editing.
- It is able to compare differences between files and merge them.
- It can search for regular expressions.
- Users can perform complicated tasks using commands.
- Users can open files in mulitple tabs.
- Users can create macros of a sequence of commands.
- Users can save sessions to start where you left off last time.

## Launching Vim

> Vim can be run directly from the command line interface or from its GUI.

Reviewer: ```suggestion
Vim can be launch directly from the terminal by running the `vim` command. The start screen will look like this:

![Vim terminal screenshot](https://raw.githubusercontent.com/Codecademy/docs/main/media/vim-terminal.png)
```

The following command will open a file named `filename.txt` in the Vim editor. If `filename.txt` does not exist, it will be created.

```bash
vim filename.txt
```

> ## Vim Screenshot

Reviewer: ```suggestion
## Vim GUI
```

> ![Vim terminal screenshot](https://raw.githubusercontent.com/Codecademy/docs/main/media/vim-terminal.png)

Reviewer: ```suggestion
```

---
Title: 'Vim'
Description: 'Vim is a free, open source text editor that is customizable text and available on many operating systems.'
Subjects:
  - 'Bash/Shell'
  - 'Developer Tools'
Tags:
  - 'Vim'
  - 'Linux'
  - 'Developer Tools'
CatalogContent:
  - 'learn-the-command-line'
  - 'paths/computer-science'
---

[Vim](https://www.vim.org/) is a text editor that is free, [open-source](https://www.codecademy.com/resources/docs/general/open-source), and based on the `vi` editor that was originally created for the Unix operating system. However, it can be run on other systems such as Windows, Mac, Linux, or Android.

Many plugins are available to add to the many features built into Vim. Developers will find it useful for working on source code or scripting files. Vim is ideal for editing configuration files when working from the [terminal](https://www.codecademy.com/resources/docs/general/terminal).

## Features

> This section describes some of the important features of the Vim editor.

Reviewer 2: ```suggestion
This section describes some important features of the Vim editor.
```

- It has text completion for faster editing.
- It is able to compare differences between files and merge them.
- It can search for regular expressions.

> - Users can perform complicated tasks using commands.

Reviewer 2: Do you mean "using keyboard shortcuts"?

- Users can open files in mulitple tabs.
- Users can create macros of a sequence of commands.
- Users can save sessions to start where you left off last time.

## Launching Vim

> Vim can be launch directly from the terminal by running the `vim` command. The start screen will look like this:

Reviewer 3: ```suggestion
Vim can be launched directly from the terminal by running the `vim` command. The start screen will look like this:
```

![Vim terminal screenshot](https://raw.githubusercontent.com/Codecademy/docs/main/media/vim-terminal.png)

The following command will open a file named `filename.txt` in the Vim editor. If `filename.txt` does not exist, it will be created.

```bash
vim filename.txt
```

> ## Vim GUI

Reviewer 2: ```suggestion
## Vim Graphical User Interface (GUI)
```
